src/memory/title_sequence_manager.rs

```diff
@@ -29,104 +47,352 @@ impl MemoryManagerUpdate for TitleSequenceManagerData {
+            Err(_) => {
+                return Ok(());
+            }
+        };
+
+        let right_selected = match class.follow_fields::<u8>(
+            singleton,
+            process,
+            module,
+            &["characterSelectionScreen", "rightButton", "selected"],
+        )? {
+            0 => false,
+            1 => true,
+            _ => false,
+        };
+
+        let mut selected = match class.follow_fields::<u64>(
+            singleton,
+            process,
+            module,
+            &["characterSelectionScreen", "selectedCharacter"],
+        ) {
+            Ok(value) => {
+                if value == 0 {
+                    PlayerPartyCharacter::None
+                } else {
+                    let selected_name_addr = class.follow_fields::<u64>(
+                        singleton,
+                        process,
+                        module,
+                        &[
+                            "characterSelectionScreen",
+                            "selectedCharacter",
+                            "characterDefinitionId",
+                        ],
+                    )?;
+
+                    let selected_name_str =
+                        process.read_pointer::<ArrayWString<16>>(selected_name_addr + 0x14)?;
+                    let selected_name = match String::from_utf16(selected_name_str.as_slice()) {
+                        Ok(value) => value,
+                        Err(_) => {
                             return Ok(());
                         }
-                        if class
-                            .follow_fields::<u8>(
-                                singleton,
-                                process,
-                                module,
-                                &["titleScreen", "quitGameButton", "selected"],
-                            )
-                            .ok()
-                            == Some(1)
-                        {
-                            self.title_menu.selected = TitleMenuOption::QuitGame
-                        }
-                    }
+                    };
+                    PlayerPartyCharacter::parse(&selected_name)
                 }
             }
+            Err(_) => PlayerPartyCharacter::None,
+        };
+
+        let character_selected = match class.follow_fields::<u8>(
+            singleton,
+            process,
+            module,
+            &["characterSelectionScreen", "characterSelected"],
+        )? {
+            0 => false,
+            1 => true,
+            _ => false,
+        };
+
+        let left = NewGameCharacter {
+            character: PlayerPartyCharacter::parse(&left_name),
+            selected: left_selected,
+        };
+
+        let right = NewGameCharacter {
+            character: PlayerPartyCharacter::parse(&right_name),
+            selected: right_selected,
+        };
+
+        if !character_selected && (right.selected && !left.selected)
```

Comment:
This logic block is a bit dense. Either rewrite to something easier to parse, or leave a comment on what it does.

Reply:
Yea, the fields on the 3 different objects work kinda strangely depending on where the cursor is - i'm not entirely sure if L375 is required - will check.

for example: right.selected and left.selected on character select entry will be `1`, if you go to relic screen and go back they both will be `0`

once you move the cursor it will update right/left selected to be inverse
character selected only updates the character to be the first record you selected, so we fall back to the right/left during that time